### ISSUE TITLE
‼️ (python): NameError: name 'SubnetSelection' is not defined

### ISSUE THREAD

**Issue by anggras** [aws/aws-cdk#21790]
Please add your +1 👍 to let us know you have encountered this
---

### Overview:

CDK Python applications throw a `NameError` error on synthesis indicating that a name is 'not defined'.

For example when trying to use `SubnetSelection`

```
lb = elbv2.ApplicationLoadBalancer(
            self, "MainLB",
            vpc=vpc,
            vpc_subnets={
                'subnet_group_name': "Ingress"
            },
            internet_facing=True
        )
```

Will throw the error `NameError: name 'SubnetSelection' is not defined`

#### Complete Error Message:

```
Traceback (most recent call last):
  File "/tmp/cdk-tmp-python/app.py", line 10, in <module>
    CdkTmpPythonStack(app, "CdkTmpPythonStack",
  File "/tmp/cdk-tmp-python/.venv/lib/python3.9/site-packages/jsii/_runtime.py", line 86, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/tmp/cdk-tmp-python/cdk_tmp_python/cdk_tmp_python_stack.py", line 24, in __init__
    lb = elbv2.ApplicationLoadBalancer(
  File "/tmp/cdk-tmp-python/.venv/lib/python3.9/site-packages/jsii/_runtime.py", line 86, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/tmp/cdk-tmp-python/.venv/lib/python3.9/site-packages/aws_cdk/aws_elasticloadbalancingv2/__init__.py", line 16381, in __init__
    props = ApplicationLoadBalancerProps(
  File "/tmp/cdk-tmp-python/.venv/lib/python3.9/site-packages/aws_cdk/aws_elasticloadbalancingv2/__init__.py", line 15357, in __init__
    vpc_subnets = SubnetSelection(**vpc_subnets)
NameError: name 'SubnetSelection' is not defined
```

### Workaround:

Currently the only workaround is to stay on `v2.38.1`

### Framework Version

2.39.0

### Language

Python

**Comment by corymhall** [comment]
@anggras thanks for reporting this issue! We've labeled this as a `p0` and have identified a fix in [JSII](https://github.com/aws/jsii/pull/3728). We will update this ticket once the fix has been released.

**Comment by ryparker** [comment]
Fix has been released in CDK [v2.39.1](https://github.com/aws/aws-cdk/releases/tag/v2.39.1).

**Comment by ryparker** [comment]
Fix has been released for v1 CDK in [1.170.1](https://www.npmjs.com/package/aws-cdk/v/1.170.1)
